src/v/cloud_storage/remote_partition.cc

```diff
@@ -527,17 +571,22 @@ class partition_record_batch_reader_impl final
             _partition->evict_reader(std::move(_reader));
             vlog(
               _ctxlog.debug,
-              "initializing new segment reader {}, next offset {}",
+              "initializing new segment reader {}, next offset {}, manifest "
+              "cursor status: {}",
               config.start_offset,
-              _next_segment_base_offset);
-            if (_next_segment_base_offset != model::offset{}) {
+              _next_segment_base_offset,
+              _view_cursor->get_status());
+            auto maybe_manifest = _view_cursor->manifest();
+            if (
+              !maybe_manifest.has_value()
```

Comment:
Should remove the not? Or perhaps we shouldn't call `maybe_manifest.value()` at L585?